webui/react/src/pages/ResourcePool/NodeElement.tsx

```diff
@@ -0,0 +1,82 @@
+import Tooltip from 'hew/Tooltip';
+import React, { PropsWithChildren, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { Resource, SlotsRecord } from 'types';
+
+import css from './NodeElement.module.scss';
+
+interface NodeElementProps {
+  name: string;
+  resources: Resource[];
+  slots?: SlotsRecord;
+  isRunning?: boolean;
+  limitSlots?: number;
+}
+
+const NodeElement: React.FC<PropsWithChildren<NodeElementProps>> = ({
+  name,
+  slots,
+  resources,
+  isRunning = true,
+  limitSlots = 0,
+}) => {
+  const [containerWidth, setContainerWidth] = useState(0);
+  const shouldTruncate = useMemo(() => name.length > 5, [name]);
+  const slotsContainer = useRef<HTMLSpanElement>(null);
+  const slotsData = useMemo(
+    () => (slots !== undefined ? Object.values(slots) : resources),
+    [slots, resources],
+  );
+  const singleSlot = slotsData.length === 1;
+  const coupleSlot = slotsData.length === 2;
```

Comment:
q: how come `coupleSlot` is `slotsData.length === 2` instead of slotsData.length >== 2;?

Reply:
could you rename the variable then?